### PR TITLE
Remove codegen_mappings.json, update gulpfile tasks

### PR DESCRIPTION
**Pull request by daschult**
In this PR, I've moved all of the contents of codegen_mappings.json to the readme.nodejs.md files in the azure-rest-api-specs repository (see https://github.com/Azure/azure-rest-api-specs/pull/2788 for the results of the move). This means that all of the new swagger specification PRs will have the latest information needed to generate Node.js packages. This also means that since the details are now in the azure-rest-api-specs repository in readme.md files, we can just point autorest at those readme files. This means that we don't need codegen_mappings.json any more. This PR also updates our gulpfile to generate our packages by iterating over your local clone of the azure-rest-api-specs repo and looking for readme.nodejs.md files. It will generate a package for each of those files that it finds.

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -6,245 +6,131 @@
...
+          nodejsReadmeFilePaths.push(folderEntryPath);
+        }
       }
-    });
-    fs.rmdirSync(path);
+    }
   }
-}
 
-function clearProjectBeforeGenerating(projectDir) {
```
Comment (amarzavery) [inline gulpfile.js]: W.r.t cleaning projects before generating there are sdks like azure-batch, azure-keyvault and datalake analytics/filesystem that have some hand written files along with the autogenerated files. We should make sure that Autorest does not accidentally delete them.

Comment (daschult) [reply gulpfile.js]: I'll verify this. Hold on.

File: gulpfile.js [hunk 2]
```diff
@@ -436,102 +234,26 @@ gulp.task('sync-deps-rollup', (cb) => {
   fs.writeFileSync('./package.json', JSON.stringify(rollupPackage, null, 2), { 'encoding': 'utf8' });
 });
 
-gulp.task('sync-package-service-mapping', (cb) => {
```
Comment (amarzavery) [inline gulpfile.js]: this task is used to update package_service_mapping.json . This file is used by the docs team to update sdk docs on docs.microsoft.com website. We need to make sure that we update this mapping once we publish packages. I think I missed letting you know about this. It is my fault.

File: gulpfile.js (hunk 1 above)
Comment (daschult) [reply gulpfile.js]: Yup, I checked and the handwritten stuff stayed put.

File: gulpfile.js (hunk 2 above)
Comment (daschult) [reply gulpfile.js]: Don't worry about it. It's not possible to tell me everything in advance, and even if you tried, I probably wouldn't remember it. This is the perfect time to learn about this. I'll fix it.

File: codegen_mappings.schema.json [hunk 3]
```diff
@@ -1,197 +0,0 @@
-{
```
Comment (RikkiGibson) [inline codegen_mappings.schema.json]: Goodbye codegen_mappings.schema.json, we hardly knew ye.

File: gulpfile.js [hunk 4]
```diff
@@ -6,245 +6,131 @@
...
-    console.log(execHelpCmd);
-    execHelp = execSync(execHelpCmd, { encoding: 'utf8' });
-    //console.debug(execHelp);
-  } catch (err) {
-    result = false;
-    console.log(`An error occurred while getting the "${version}" of autorest and executing "autorest --help":\n ${util.inspect(err, { depth: null })}.`);
-  }
-  return result;
-}
+function findReadmeNodejsMdFilePaths(azureRestAPISpecsRoot) {
```
Comment (RikkiGibson) [inline gulpfile.js]: Can't we use something like node-glob for this?

File: gulpfile.js [hunk 5]
```diff
@@ -6,245 +6,131 @@
...
-        }
-        generateProject(proj[key], specRoot, usingAutoRestVersion);
-      } else {
-        iterateProject(proj[key], specRoot, usingAutoRestVersion);
+      try {
+        console.log('Executing command:');
+        console.log('------------------------------------------------------------');
+        console.log(cmd);
+        console.log('------------------------------------------------------------');
+        const result = execSync(cmd, { encoding: 'utf8' });
```
Comment (RikkiGibson) [inline gulpfile.js]: This PR might not be the time, but I'd really like to drill into why we can't do generation in parallel.

Comment (daschult) [reply gulpfile.js]: I'm sure we can, but you're totally right that this isn't the PR for it.

File: gulpfile.js (hunk 4 above)
Comment (daschult) [reply gulpfile.js]: Yeah, we probably can. I can look into that.

Comment (daschult) [reply gulpfile.js]: Done.

File: gulpfile.js (hunk 2 above)
Comment (daschult) [reply gulpfile.js]: Done.

**Pull request merged**